lib/rules/prefer-aria-over-title-attribute.ts

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { Rule } from "eslint";
+import { JSXAttribute, JSXOpeningElement, Literal } from "estree-jsx";
+import { elementType, hasProp } from "jsx-ast-utils";
+import { hasAssociatedLabelViaAriaLabelledBy } from "../util/labelUtils";
+import { hasNonEmptyProp } from "../util/hasNonEmptyProp";
+import { hasToolTipParent } from "../util/hasTooltipParent";
+import { hasTextContentChild } from "../util/hasTextContentChild";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const applicableComponents = ["Button"];
+
+const rule: Rule.RuleModule = {
+    meta: {
+        // possible error messages for the rule
+        messages: {
+            preferAria: `Prefer aria over the title attribute for accessible labelling: ${applicableComponents.join(", ")}`
+        },
+        type: "suggestion", // `problem`, `suggestion`, or `layout`
+        docs: {
+            description:
+                "The title attribute is not consistently read by screen readers, and its behavior can vary depending on the screen reader and the user's settings.",
+            recommended: true
+        },
+        fixable: "code", // Or `code` or `whitespace`
+        schema: [] // Add a schema if the rule has options
+    },
+
+    create(context) {
+        return {
+            // visitor functions for different types of nodes
+            JSXElement(node: any) {
+                const openingElement = node.openingElement as JSXOpeningElement;
```

Comment:
Instead of casting directly, is it possible to find the type of the node and declare the type?

Reply:
the JSXElement function is a non-standard function and thus the types are not defined

Reply:
HI @ajayagr , I think there might be typings in this package? import { TSESTree } from '@typescript-eslint/types';

function JSXElement(node: TSESTree.JSXElement) {
    const openingElement = node.openingElement as TSESTree.JSXOpeningElement;
    // Now you can safely work with the openingElement
}

Reply:
this is available but then it is failing against the util function imported from jsx-ast-utils package. This is why manual typecasting is required. Although you bring a good point that the manual typecasting can be performed at the function call level instead of at the root. will fix it and update accordingly.